genesyscloud/telephony_provider_edges_trunkbasesettings/genesyscloud_telephony_providers_trunkbasesettings_proxy.go

```diff
@@ -0,0 +1,193 @@
+package telephony_provider_edges_trunkbasesettings
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"terraform-provider-genesyscloud/genesyscloud/util"
+	"time"
+
+	rc "terraform-provider-genesyscloud/genesyscloud/resource_cache"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/mypurecloud/platform-client-sdk-go/v143/platformclientv2"
+)
+
+var internalProxy *trunkbaseSettingProxy
+var trunkBaseCache = rc.NewResourceCache[platformclientv2.Trunkbase]()
+
+// Type definitions for each func on our proxy so we can easily mock them out later
+type getTrunkBaseSettingByIdFunc func(ctx context.Context, p *trunkbaseSettingProxy, id string) (*platformclientv2.Trunkbase, *platformclientv2.APIResponse, error)
+type getAllTrunkBaseSettingsFunc func(ctx context.Context, p *trunkbaseSettingProxy, name string) (*[]platformclientv2.Trunkbase, *platformclientv2.APIResponse, error)
+type createTrunkBaseSettingFunc func(ctx context.Context, p *trunkbaseSettingProxy, trunkBaseSetting platformclientv2.Trunkbase) (*platformclientv2.Trunkbase, *platformclientv2.APIResponse, error)
+type updateTrunkBaseSettingFunc func(ctx context.Context, p *trunkbaseSettingProxy, id string, trunkBaseSetting platformclientv2.Trunkbase) (*platformclientv2.Trunkbase, *platformclientv2.APIResponse, error)
+type deleteTrunkBaseSettingFunc func(ctx context.Context, p *trunkbaseSettingProxy, id string) (*platformclientv2.APIResponse, error)
+
+type trunkbaseSettingProxy struct {
+	clientConfig *platformclientv2.Configuration
+	edgesApi     *platformclientv2.TelephonyProvidersEdgeApi
+
+	getTrunkBaseSettingByIdAttr getTrunkBaseSettingByIdFunc
+	getAllTrunkBaseSettingsAttr getAllTrunkBaseSettingsFunc
+	createTrunkBaseSettingAttr  createTrunkBaseSettingFunc
+	updateTrunkBaseSettingAttr  updateTrunkBaseSettingFunc
+	deleteTrunkBaseSettingAttr  deleteTrunkBaseSettingFunc
+	trunkBaseCache              rc.CacheInterface[platformclientv2.Trunkbase]
+}
+
+// initializes the  proxy with all of the data needed to communicate with Genesys Cloud
+func newTrunkBaseSettingProxy(clientConfig *platformclientv2.Configuration) *trunkbaseSettingProxy {
+	edgesApi := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(clientConfig)
+	return &trunkbaseSettingProxy{
+		clientConfig:                clientConfig,
+		edgesApi:                    edgesApi,
+		getTrunkBaseSettingByIdAttr: getTrunkBaseSettingByIdFn,
+		createTrunkBaseSettingAttr:  createTrunkBaseSettingFn,
+		updateTrunkBaseSettingAttr:  updateTrunkBaseSettingFn,
+		deleteTrunkBaseSettingAttr:  deleteTrunkBaseSettingFn,
+		getAllTrunkBaseSettingsAttr: getAllTrunkBaseSettingsFn,
+		trunkBaseCache:              trunkBaseCache,
+	}
+}
+
+// getTeamProxy acts as a singleton to for the internalProxy.  It also ensures
```

Comment:
can rename this to getTrunkBaseSettingProxy*

Reply:
Fixed